client/src/App.js

```diff
@@ -1,12 +1,230 @@
-import React, {useEffect, useState} from 'react';
-import logo from './logo.svg';
-import Ticket from './components/Ticket'
+import React, { useState, useEffect } from 'react';
+import InputLabel from '@material-ui/core/InputLabel';
+import { TextField } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import NativeSelect from '@material-ui/core/NativeSelect';
+import axios from 'axios';
+import Ticket from './components/Ticket';
+import Header from './components/Header';
 import './App.css';
 
 function App() {
+  const [TicketArr, setTicketArr] = useState([]);
+  const [sortByTime, setSortByTime] = useState([[], 'default']);
+  const [dateRange, setDateRange] = useState(
+    {
+      from: (new Date(2017, 11, 1)).getTime(),
+      to: Date.now(),
+    },
+  );
+  const [showFinished, setShowFinished] = useState(false);
+  // const ActiveTickets = sortByTime[0].filter((ticket) => !ticket.done && !ticket.hidden);
+  // const HandeledTickets = sortByTime[0].filter((ticket) => ticket.done && !ticket.hidden);
+  let ActiveTickets =[], HandeledTickets = [],hiddenCounter1=0 , hiddenCounter2 =0;
+  sortByTime[0].forEach(ticket => {
+    if(ticket.hidden && !ticket.done){
+      hiddenCounter1++;
+    }else if(ticket.hidden && ticket.done){
+      hiddenCounter2++;
+    }
+      if(ticket.done){
+        HandeledTickets.push(ticket);
+      }else{
+        ActiveTickets.push(ticket);
+      }
+  })
+  //const hiddenCounter1 = sortByTime[0].length - (ActiveTickets.length + HandeledTickets.length);
+
+  function oldOrNew(val, arr) {
+    const arrCopy = arr ? arr.slice() : TicketArr.slice();
+    const newArr = arrCopy.filter((ticket) => ticket.creationTime < dateRange.to && ticket.creationTime > dateRange.from);
+    if (val === '0') {
+      newArr.sort((a, b) => a.creationTime - b.creationTime);
+      setSortByTime([newArr, '0']);
+    } else if (val === '1') {
+      newArr.sort((a, b) => b.creationTime - a.creationTime);
+      setSortByTime([newArr, '1']);
+    } else {
+      setSortByTime([newArr, 'default']);
+    }
+  }
+
+  useEffect(() => {
+    axios.get('/api/tickets').then((response) => { setSortByTime([response.data, 'default']); setTicketArr(response.data); });
+  }, []);
+
+  useEffect(() => {
+    oldOrNew(sortByTime[1].toString());
+  }, [dateRange, TicketArr]);
+
+  function hideItem(id) {
+    const newArr = sortByTime[0].slice();
+    newArr.forEach((ticket, i) => {
+      if (ticket.id === id) {
+        newArr[i].hidden = true;
+        setSortByTime([newArr, sortByTime[1]]);
```

Comment:
Very bad pattern. you are trying to find the correct ticket, `forEach` is not your best option. and even if it was, you should do something with that ticket when the loop ends and not in the middle of it.

* look for other places in your code you used this pattern and fix them
* 👨🏻‍🏫 - prepare 5 min explanation